src/context.rs

```diff
@@ -192,15 +195,41 @@ pub struct Context<'a, 'b: 'a, 's> {
 
 ///A reader for a request body.
 pub struct BodyReader<'a, 'b: 'a> {
-    request: HttpReader<&'a mut BufReader<&'b mut NetworkStream>>
+    reader: HttpReader<&'a mut BufReader<&'b mut NetworkStream>>,
+    is_multipart: bool,
```

Comment:
It doesn't really work, no. You might get a multipart request that omits the boundary but I would consider it malformed.